### PR TITLE
fix(console) ignore class methods in console.log output

### PR DESCRIPTION
**Pull request by almmiko**
### What does this PR do?

Fixes https://github.com/oven-sh/bun/issues/4223

### CODE REVIEW

File: src/bun.js/bindings/exports.zig [hunk 1]
```diff
@@ -1852,6 +1852,10 @@ pub const ZigConsoleClient = struct {
                     if (key.eqlComptime("constructor")) return;
                     if (key.eqlComptime("call")) return;
 
+                    if (value.jsType().isFunction()) {
```
Comment (Jarred-Sumner) [inline src/bun.js/bindings/exports.zig]: What happens if it's a function defined on the instance but not on the prototype? 

```js
class FooBar {
  constructor() {
    this.abc = () => {};
  }
  method() {
    return 42;
  }
}
```

In node, this logs
```js
FooBar { abc: [Function (anonymous)] }
```

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
This change likely needs to be done in `forEachProperty` C++ binding and not in the Zig code as we do not have the information to know whether the properties are the prototype's properties or the instance.

image

**Comment by almmiko** [comment]
Hey @Jarred-Sumner thanks for highlighting the possible place where the change should be made.

**Review by jdalton — APPROVED** [review]

**Comment by Jarred-Sumner** [comment]
Closing due to bitrot

**Pull request closed without merging**